plotly/offline/offline.py

```diff
@@ -236,13 +280,32 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
 
     display(HTML(plot_html))
 
+    if image:
+        if image not in __IMAGE_FORMATS:
+            raise ValueError('The image parameter takes only the '
+                             'following format types: `png`, `jpeg`, '
+                             '`webp`, `svg`')
+        # if the check passes then download script injection will commence.
+        # Write script to download image of the plot
+        script = image_download_script('iplot').format(format=image,
+                                                       width=image_width,
+                                                       height=image_height,
+                                                       filename=filename,
+                                                       plot_id=plotdivid)
+        # allow time for the plot to draw
+        import time
+        time.sleep(1)
+        # inject code to download an image of the plot
+        display(HTML(script))
+
 
 def plot(figure_or_data,
          show_link=True, link_text='Export to plot.ly',
          validate=True, output_type='file',
          include_plotlyjs=True,
-         filename='temp-plot.html',
-         auto_open=True):
+         filename='temp-plot.html', auto_open=True,
```

Comment:
This won't flake if you don't supply a filename *twice*, right? Will we end up with `temp-plot.html (1)` or something? Or, does the OS just handle it?